### DISCUSSION TITLE
Want to learn about encoding/decoding to work on a new File format project

### DISCUSSION THREAD

**Discussion by leunoia** [enzo1982/freac#378] [General Discussion]
As the title says, I am interesting and learning about how encoding/decoding works, so I can develop a new encoding method that would meet specific requirements that I need ( I know that is a little vague). I was wondering if anyone can point me in a direction to where I can learn about how encoding and decoding is done on a programmatic level (preferably in C++). I have been spending a few hours reading over the source code but I don't I feel I have gotten a solid grasp as to how the process works exactly. 

I know this may be a bit of a weird post so I apologize if it isn't exactly relevant. 

Best,
Preston

**Comment by Freddiefora** [comment]
If you develop a new encode/decode chain, do you mean for audio compression? There is some information on the Monkeys Audio site under the 'theory tab'  https://monkeysaudio.com/theory.html   but why would you want to do this? Also, have a look at the WavPack site  https://www.wavpack.com/wavpack_doc.html  to get an idea of all the good work that has been undertaken here. I may have misunderstood your intention; perhaps you could expand a little on what you wish to achieve.

**Comment by Lee-Carre** [comment]
There's a wealth of technical information at [Xiph](//www.xiph.org/) (see, also, their [wiki](//wiki.xiph.org/)). Xiph is the home of many libre-codecs, some of which have beceme quite popular.
